lib/index.js

```diff
@@ -290,3 +299,74 @@ internals.stop = async (server) => {
 
 internals.state = (realm) => Toys.state(realm, 'schwifty');
 internals.rootState = (realm) => Toys.rootState(realm, 'schwifty');
+
+internals.migrateDiffCmd = async (srv, args, rootDir, ctx) => {
+
+    const SchwiftyMigrateDiff = SchwiftyMigrateDiffHelper.get();
+
+    if (!SchwiftyMigrateDiff) {
+        throw new ctx.DisplayError('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+    }
+
+    const definition = {
+        mode: {
+            description: 'Set mode to "create" or "alter" (defaults to "alter") -- when set to create the migration file ignores changes that will delete or mutate existing columns/tables'
+        },
+        dir: {
+            description: 'Specifies the directory in which to write the migration file, defaults to the migrationsDir Schwifty receives at registration'
+        }
+    };
+
+    const parsedArgs = Bossy.parse(definition, { argv: args });
+
+    if (parsedArgs instanceof Error) {
+        throw new ctx.DisplayError(parsedArgs.message + Os.EOL + Bossy.usage(definition));
+    }
+
+    Hoek.assert(parsedArgs._ && parsedArgs._.length !== 0, new ctx.DisplayError('A name must be specified for the generated migration'));
+    Hoek.assert(parsedArgs._.length === 1, new ctx.DisplayError(Bossy.usage(definition)));
+
+    const [migrationName] = parsedArgs._;
+
+    if (parsedArgs.mode) {
+        Hoek.assert(parsedArgs.mode === 'create' || parsedArgs.mode === 'alter', new ctx.DisplayError(Bossy.usage(definition)));
+    }
+
+    const { realmsWithMigrationsDir } = internals.state(srv.realm).collector;
+
+    const realmsWithMigrationsInRoot = Array.from(realmsWithMigrationsDir).reduce((collector, realm) => {
```

Comment:
Sometimes a `reduce()` is easier to read as a `map()` plus `filter()` :)

Reply:
(Or just a `filter()`, as in this case)

Reply:
D'oohhhh I keep getting trippy on which of these array funcs are the best to pick!

Reply:
Using them is super fun tho

Reply:
I think it helps to remember that `map()` and `filter()` are just special cases of `reduce()`.